Foundation/NSURLSession/NSURLSession.swift

```diff
@@ -225,7 +226,9 @@ open class URLSession : NSObject {
         identifier = nextSessionIdentifier()
         self.workQueue = DispatchQueue(label: "URLSession<\(identifier)>")
         self.taskAttributesIsolation = DispatchQueue(label: "URLSession<\(identifier)>.taskAttributes", attributes: DispatchQueue.Attributes.concurrent)
-        self.delegateQueue = queue ?? OperationQueue()
+        let operationQueue = OperationQueue()
```

Comment:
We shouldn't create an `OperationQueue` unless we need to. This creates one in all cases.  How about this:

```swift
if queue != nil {
    self.delegateQueue = queue
} else {
    self.delegateQueue = OperationQueue()
    self.delegateQueue.maxConcurrentOperationCount = 1
}
```